man/coap_block.txt.in

```diff
@@ -0,0 +1,457 @@
+used for releasing the data when the body transfer is complete.  It also adds
+in the appropriate CoAP options such as BLOCK2, SIZE2 and ETAG to handle
+Block-Wise transfer if the data does not fit into a single PDU.
+_resource_, _query_, _session_, _request_, _response_ and _token_ are the same
+parameters as in the called resource handler that invokes
+*coap_add_data_large_response*(). If _etag_ is 0, then a unique ETag value will
+be generated, else is the ETag value to use.
+The _media_type_ is for the format of the _data_ and _maxage_ defines the
+lifetime of the response.  If _maxage_ is set to -1,  then the MAXAGE option
+does not get included.  This function must only be called once per _pdu_.
+The application handler for the resource is only called once instead of
+potentially multiple times.
+
+The *coap_get_data_large*() function is used abstract from the _pdu_
+information about the received data by updating _length_ with the length of
+data available, _data_ with a pointer to where the data is located, _offset_
+with where this block of data starts and _total_ with the total amount of data.
+_offset_ will always be zero if block_mode includes COAP_BLOCK_SINGLE_BODY.
+All of the body's data has been received if "_offset_ + _length_ == _total_".
+
+*NOTE:* _total_ is potentially only an indication of the total size of the
+body and is only exact when all of the data has been received.
+
+The *coap_block_build_body*() function is used to re-assemble the received
+data as returned by *coap_get_data_large*() into a single blob of data. Data
+from _data_ of length _length_ starting from offset _offset_ is added to
+_body_data_.  The resultant state of _body_data_ is returned. If _body_data_
+is NULL, or _total_ is larger than the current size of _body_data_, then
+_body_data_ is re-allocated and returned.  If there is an error, _body_data_
+gets de-allocated.
+
+The *coap_send_large*() function is used to initiate the transmission of a
+request type _pdu_ associated with the _session_ in the same way as
+*coap_send*() but also sets up a mechanism for handling a (potentially) large
+response of data that spans multiple blocks.  To get the start of the data
+offset and total size, *coap_get_data_large*() needs to be called in the
+response handler. If the request includes the OBSERVE option with the observe
+value set to 0, then any response that are observe triggered are also handled.
+
+If _block_mode_ (as set by *coap_context_set_block_mode*()) includes
+COAP_BLOCK_SINGLE_BODY and *coap_send_large*() is used, then the response
+handler will only get called once with the entire body containing the data
+from all of the individual blocks. If there is a change of data during the
+blocks receipt (e.g. ETag value changes), then the entire set of data is
+re-requested and the partial body dropped.
+
+The *coap_cancel_observe*() function can be used to cancel an observe request
```

Comment:
I wonder if this is the right manual to include `coap_cancel_observe()`. Maybe `coap_observe.txt.in` would be more appropriate?

Reply:
I have moved this over to `coap_observe.txt.in` and clarified further which are server and which are client functions.